cpp/src/cones.cpp

```diff
@@ -324,9 +326,12 @@ LinearOperator _dprojection(const Vector &x, ConeType type, bool dual) {
     return _dprojection_soc(x);
   } else if (type == PSD) {
     return _dprojection_psd(x);
-  } else {
-    assert(type == EXP);
+  } else if (type == EXP) {
     return _dprojection_exp(x, dual);
+  } else if (type == EXP_DUAL) {
+    return _dprojection_exp(x, !dual);
+  } else {
+    throw "Cone not supported.";
```

Comment:
`throw invalid_argument("Cone not supported")`? https://docs.microsoft.com/en-us/cpp/standard-library/stdexcept?view=vs-2019